saleor/checkout/complete_checkout.py

```diff
@@ -0,0 +1,178 @@
+from typing import List, Optional, Tuple
+
+from django.core.exceptions import ValidationError
+from django.db import transaction
+
+from ..account.error_codes import AccountErrorCode
+from ..account.models import User
+from ..core.exceptions import InsufficientStock
+from ..core.taxes import TaxError
+from ..core.utils.url import validate_storefront_url
+from ..discount import DiscountInfo
+from ..discount.models import NotApplicable
+from ..order import models as order_models
+from ..order.models import Order
+from ..payment import PaymentError, gateway
+from ..payment.models import Payment, Transaction
+from ..payment.utils import store_customer_id
+from . import models
+from .checkout_cleaner import clean_checkout_payment, clean_checkout_shipping
+from .error_codes import CheckoutErrorCode
+from .utils import abort_order_data, create_order, get_order, prepare_order_data
+
+
+@transaction.atomic
+def prepare_checkout(
```

Comment:
I'm not sure what's the responsibility of this function, the name "prepare_checkout" doesn't explain much, there is also no docstring here.

Reply:
I've added the docstring. See if it is clear enough now